### PR TITLE
Partial feature release - shared objects and structure normalization

### PR DESCRIPTION
**Pull request by gklundt**
Added my domain data classes, repository interfaces and stub code for exercise fragment and edit screen.  Common resources were touched so I wanted to get this merged before more conflicts are generated.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,8 @@
 /local.properties
 /.idea/workspace.xml
 /.idea/libraries
+/.idea/codeStyleSettings.xml
+
```
Comment (gklundt) [inline .gitignore]: This and app.iml are giving me fits.  I guess it's locally generated crap.

**Comment by symbyte** [comment]
Exercise List fragment crashes when launching from app drawer.

Exception:
10-26 07:12:14.735 13950-13950/edu.uco.schambers4.octane E/AndroidRuntime: java.lang.ClassCastException: edu.uco.schambers4.octane.Activities.MainActivity@28c68c2f must implement OnFragmentInteractionListener

**Comment by symbyte** [comment]
Pushed a fix, but I guess I'm not able push commits to your pull request directly without creating another one, just going to merge this for now in effort to avoid conflicts like you mentioned.

**Comment by gklundt** [comment]
Yeah, the feature is not complete (bad practice), but it was getting late
and I wanted to get the structure and common objects up before you started
working on your next bit.  I'll push in another request today some time
that actually works.

**Comment by symbyte** [comment]
Ok cool, sounds good.

**Pull request merged**